### PR TITLE
Screen based login UI and tons of fixes

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
**Changes**

- Discovery servers are not shown by default
- Discovery servers are shown in the "add server" tab
- Add button to add server to the same tab
- Add help text in add server tab
- Servers are now sorted on last use
- Users are now sorted on last use
- Servers now update when a new server is added
- Fixed some behavior issues
- Simplified authentication and server repositories
- Miscellaneous fixes

**Issues**

Part of #612

### CODE REVIEW

**Comment by crobibero** [comment]
The only thing I’ve really noticed is it seems like focus gets lost in the discovered servers pane when there are none.

**Comment by thornbill** [comment]
I was coming here to comment this, but it seems I was too slow. :rofl:

**Comment by nielsvanvelzen** [comment]
Rebased this branch so I could deploy it to my television. no changes made.

**Comment by thornbill** [comment]
Screenshot_1614204329

Some feedback on this screen:

1. The subheadings for "Discovered servers" and "Connect manually" can probably be removed.
2. The "Need help" section needs some additional spacing. Maybe center it vertically or align it to the bottom?
3. Maybe the "Add Server" title at the bottom should be changed to something else since we have a button to "Add Server" that opens the dialog for entering a server manually. "Connect" perhaps?

File: app/src/main/java/org/jellyfin/androidtv/auth/ServerRepository.kt [hunk 1]
```diff
@@ -31,95 +33,36 @@ class ServerRepositoryImpl(
...
-		getServers(discovery, stored).collect { servers ->
-			emit(servers.map { server ->
-				if (server.id !in userFlows)
-					userFlows[server.id] = getUsers(server)
-
-				server to userFlows[server.id]!!
-			}.toMap())
-		}
-	}
+	override suspend fun mirateLegacyCredentials() = legacyAccountMigration.migrate()
```
[the remark this answers is not in the document]
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/auth/ServerRepository.kt:65]: Mirating all the things

**Review by thornbill — APPROVED** [review]

**Pull request merged**